### PR TITLE
Added new module mail_id_harrvest as per issue #143 also solved issue #200

### PR DESCRIPTION
**Pull request by itsdivyanshjain**
#### Checklist
- [x] I have followed the [Contributor Guidelines](https://github.com/zdresearch/OWASP-Nettacker/wiki/Developers#contribution-guidelines).
- [x] I have added the relevant documentation.
- [x] My branch is up-to-date with the Upstream master branch.

#### Changes proposed in this pull request
I have added tool mail_id_harvest for better passive information gathering and it has no conflict with master.
-right now google is only making request but there is few exception is there which are been added soon as i am tackling that issue

#### Your development environment
- OS: `Parrot`
- OS Version: `4.19.0`
- Python Version: `3.x`

### CODE REVIEW

File: lib/scan/mail_id_harvest/engine.py [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'mail_id_harvest_scan',
+                               'DESCRIPTION': messages(language, "domain_found").format(len(mails), ", ".join(mails) if len(
+                                   _values) > 0 else "None"), 'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id,
+                               'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+            
+    else:
+        warn(messages(language, "input_target_error").format(
+            'mail_id_harvest_scan', target))
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 8
           

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  19
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

File: lib/scan/mail_id_harvest/engine.py [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import time
+import requests
+import socks
+import socket
+import json
+import re
+from bs4 import BeautifulSoup as BS
+from core._time import now
+from xml.etree import ElementTree as ET
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Using ElementTree to parse untrusted XML data is known to be vulnerable to XML attacks. Replace ElementTree with the equivalent defusedxml package, or make sure defusedxml.defuse_stdlib() is called.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+                    for item in information:
+                        if item.endswith(target) and item not in mails:
+                            mails.append(item)
+        
+        #we get the mails list through search now appending in the database                    
+        for mail in mails:
+            try:
+                if verobose_level>3:
+                    info(messages(languages,"done").format(mail))
+            except:
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Try, Except, Pass detected.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py (hunk 2 above)
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Unused ElementTree imported from xml.etree as ET](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py (hunk 3 above)
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [No exception type(s) specified](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+import socks
+import socket
+import json
+import re
+from bs4 import BeautifulSoup as BS
+from core._time import now
+from xml.etree import ElementTree as ET
+from core.alert import *
+from core.targets import target_type
+from core.targets import target_to_host
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Unused target_to_host imported from core.targets](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 5]
```diff
@@ -0,0 +1,117 @@
...
+            __log_into_file(log_in_file, 'a', data, language)
+            
+            
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'mail_id_harvest_scan',
+                               'DESCRIPTION': messages(language, "domain_found").format(len(mails), ", ".join(mails) if len(
+                                   _values) > 0 else "None"), 'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id,
+                               'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+                    mail_pattern=re.compile(
+                        '[a-zA-Z0-9.\-_+#~!$&\',;=:]+' +
+                        '@' +
+                        '[a-zA-Z0-9.-]*' + target)
+                    information=mail_pattern.findall(result.text)
+                    for item in information:
+                        if item.endswith(target) and item not in mails:
+                            mails.append(item)
+        
+        #we get the mails list through search now appending in the database
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 6
           

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  19
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

File: lib/scan/mail_id_harvest/engine.py [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+            
+            
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 
+                'TYPE': 'mail_id_harvest_scan',
+                'DESCRIPTION': messages(language, "domain_found").format(len(mails), 
+                ", ".join(mails) if len(_values) > 0 else "None"), 'TIME': now(), 
+                'CATEGORY': "scan", 'SCAN_ID': scan_id,'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+                                   'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id,
+                                   'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+            
+            
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 
+                'TYPE': 'mail_id_harvest_scan',
+                'DESCRIPTION': messages(language, "domain_found").format(len(mails), 
+                ", ".join(mails) if len(_values) > 0 else "None"), 'TIME': now(),
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+            #info(messages(language,"choose_scan_method").format(mail))
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '',
+                                   'TYPE': 'mail_id_harvest_scan', 'DESCRIPTION':mail,
+                                   'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id,
+                                   'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+            
+            
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '',
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+                                   'TYPE': 'mail_id_harvest_scan', 'DESCRIPTION':mail,
+                                   'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id,
+                                   'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+            
+            
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 
+                'TYPE': 'mail_id_harvest_scan',
+                'DESCRIPTION': messages(language, "domain_found").format(len(mails),
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 11]
```diff
@@ -0,0 +1,116 @@
...
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 
+                'TYPE': 'mail_id_harvest_scan',
+                'DESCRIPTION': messages(language, "domain_found").format(len(mails), 
+                ", ".join(mails) if len(_values) > 0 else "None"), 'TIME': now(), 
+                'CATEGORY': "scan", 'SCAN_ID': scan_id,'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+            
+    else:
+        warn(messages(language, "input_target_error").format('mail_id_harvest_scan', target))
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 12]
```diff
@@ -0,0 +1,116 @@
...
+                        '[a-zA-Z0-9.\-_+#~!$&\',;=:]+' +
+                        '@' +
+                        '[a-zA-Z0-9.-]*' + target)
+                    information=mail_pattern.findall(result.text)
+                    for item in information:
+                        if item.endswith(target) and item not in mails:
+                            mails.append(item)
+        
+                         
+        for mail in mails:
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 5
           

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  19
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  7
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

File: lib/scan/mail_id_harvest/engine.py [hunk 13]
```diff
@@ -0,0 +1,108 @@
...
+                for result in results:
+                    mail_pattern=re.compile(
+                        '[a-zA-Z0-9.\-_+#~!$&\',;=:]+' +
+                        '@' +
+                        '[a-zA-Z0-9.-]*' + target)
+                    information=mail_pattern.findall(result.text)
+                    
+                    for item in information:
+                        if item.endswith(target) and item not in mails:
+                            mails.append(item)
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 14]
```diff
@@ -0,0 +1,108 @@
...
+                    info(messages(languages,"done"))
+            except Namerror:
+                info(messages(languages,"done"))
+            #info(messages(language,"choose_scan_method").format(mail))
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'mail_id_harvest_scan', 'DESCRIPTION':mail, 'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'mail_id_harvest_scan', 'DESCRIPTION': messages(language, "domain_found").format(len(mails), ", ".join(mails) if len(_values) > 0 else "None"), 'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id,'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 15]
```diff
@@ -0,0 +1,108 @@
...
+            #info(messages(language,"choose_scan_method").format(mail))
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'mail_id_harvest_scan', 'DESCRIPTION':mail, 'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'mail_id_harvest_scan', 'DESCRIPTION': messages(language, "domain_found").format(len(mails), ", ".join(mails) if len(_values) > 0 else "None"), 'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id,'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language) 
+  
+    else:
+        warn(messages(language, "input_target_error").format('mail_id_harvest_scan', target))
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 16]
```diff
@@ -0,0 +1,108 @@
...
+                        '[a-zA-Z0-9.\-_+#~!$&\',;=:]+' +
+                        '@' +
+                        '[a-zA-Z0-9.-]*' + target)
+                    information=mail_pattern.findall(result.text)
+                    
+                    for item in information:
+                        if item.endswith(target) and item not in mails:
+                            mails.append(item)   
+                            
+        for mail in mails:
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 17]
```diff
@@ -0,0 +1,108 @@
...
+            except Namerror:
+                info(messages(languages,"done"))
+            #info(messages(language,"choose_scan_method").format(mail))
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'mail_id_harvest_scan', 'DESCRIPTION':mail, 'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language)
+
+        if verbose_level is not 0:
+            data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': '', 'TYPE': 'mail_id_harvest_scan', 'DESCRIPTION': messages(language, "domain_found").format(len(mails), ", ".join(mails) if len(_values) > 0 else "None"), 'TIME': now(), 'CATEGORY': "scan", 'SCAN_ID': scan_id,'SCAN_CMD': scan_cmd}) + "\n"
+            __log_into_file(log_in_file, 'a', data, language) 
+
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  19
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

File: lib/scan/mail_id_harvest/engine.py [hunk 18]
```diff
@@ -0,0 +1,107 @@
...
+                    mail_pattern=re.compile(
+                        '[a-zA-Z0-9.\-_+#~!$&\',;=:]+' +
+                        '@' +
+                        '[a-zA-Z0-9.-]*' + target)
+                    information=mail_pattern.findall(result.text)
+                    
+                    for item in information:
+                        if item.endswith(target) and item not in mails:
+                            mails.append(item)
+
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  19
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

File: lib/scan/mail_id_harvest/engine.py [hunk 19]
```diff
@@ -0,0 +1,107 @@
...
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  19
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

File: lib/scan/mail_id_harvest/engine.py [hunk 20]
```diff
@@ -0,0 +1,107 @@
...
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 2
           

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  19
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

File: lib/scan/mail_id_harvest/engine.py [hunk 21]
```diff
@@ -0,0 +1,99 @@
...
+                results=parse.select(search_engines[engine])
+                for result in results:
+                    mail_pattern=re.compile(
+                        '[a-zA-Z0-9.\-_+#~!$&\',;=:]+' +
+                        '@' +
+                        '[a-zA-Z0-9.-]*' + target)
+                    information=mail_pattern.findall(result.text)
+                    for item in information:
+                        if item.endswith(target) and item not in mails:
+                            mails.append(item)
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

File: lib/scan/mail_id_harvest/engine.py [hunk 22]
```diff
@@ -0,0 +1,99 @@
...
+    
+def search_platform(target,search):
+    '''
+    Args:
+        target = Domain
+        search=search engines for parsing query
+    
+    Returns:
+        the search request url dictionary
+
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  19
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

File: lib/scan/mail_id_harvest/engine.py [hunk 23]
```diff
@@ -0,0 +1,96 @@
...
+def extra_requirement_dict():
+    return {}
+    
+def search_platform(target,search):
+    '''
+    Args:
+        target = Domain
+        search=search engines for parsing query
+
```
Comment (Ali-Razmjoo) [inline lib/scan/mail_id_harvest/engine.py]: Codacy Issue found: [Trailing whitespace](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  19
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  5
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

**Comment by shaddygarg** [comment]
@itsdivyanshjain, please look into the PEP8 standards for writing code.

**Comment by Ali-Razmjoo** [comment]
@shaddygarg can you review and test this one also?

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- lib/scan/mail_id_harvest/engine.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  20
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  22
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  22
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

**Comment by Ali-Razmjoo** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Complexity increasing per file
==============================
- lib/scan/mail_id_harvest/engine.py  22
         

Clones added
============
- lib/scan/mail_id_harvest/engine.py  6
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5096791&bid=16765901)

**Comment by itsdivyanshjain** [comment]
I also solved issue #200 for py3dns by changing it to version 3.2.1 and also do review for mail_id_harvest module

**Comment by itsdivyanshjain** [comment]
closing due to high time complexity of a code

**Pull request closed without merging**